src/here-studio.ts

```diff
@@ -0,0 +1,298 @@
+    clonedProjectData.rot = currentUsersToken;
+
+    let newProjectResponse = await execute(uri, "POST", cType, clonedProjectData, options.token);
+
+    let newProjectBody = newProjectResponse.body;
+    let studioBaseURL = "https://studio.here.com";
+    let clonedProjectURL = studioBaseURL+"/studio/project/"+newProjectBody.id;
+
+    //Viewer URL - /https://studio.here.com/viewer/?project_id=b1e3a9d4-116b-407b-b99f-17fbdf48d405
+    let viewerURL = studioBaseURL+"/viewer/?project_id="+newProjectBody.id
+    if (newProjectResponse.statusCode == 201) {
+        console.log("\n**** Successfully cloned ******")
+        console.log("Project cloned in your studio account : "+clonedProjectURL)
+        console.log("Viewer URL : "+viewerURL)
+    }
+}
+
+//Will capture URL Query string parameters from URL
+function getParameterByName (name: any, url: any) {
+    if (!url) return null;
+    name = name.replace(/[\[\]]/g, '\\$&');
+    var regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)'),
+        results = regex.exec(url);
+    if (!results) return null;
+    if (!results[2]) return '';
+    return decodeURIComponent(results[2].replace(/\+/g, ' '));
+}
+
+async function showProject (id : any, options: any) {
+    const opn = require("opn");
+    opn(
+        "https://xyz.here.com/viewer/?project_id="+id
+        , { wait: false });
+}
+
+async function deleteProject  (id : any, options: any) {
+    console.log("Deleting project : "+id)
+
+    //If project exists send a DELETE request for that projectID
+    const uri = "/project-api/projects/"+id;
+    const cType = "";
+    let response = await execute (uri, "DELETE", cType, "", options.token);
+
+    if (response && response.statusCode === 204) {
+        console.log("Successfully deleted project.")
+    }
+    else {
+        console.log("Unable to delete project having project-id: "+id)
+    }
+}
+
+
+/**
+ * Will list all the projects for the given user in below format
+ *
+ * @param options
+ */
+export async function listProjects (options: any) {
+    console.log("Please wait; Fetching your list of projects...")
+    const uri = "/project-api/projects";
+    const cType = "";//"application/json";//
+    let response = await execute(uri, "GET", cType, "", options.token);
+    let body = JSON.parse(response.body)
+    if (response.body.length == 0) {
+        console.log("No xyz projects found");
+    } else {
+        let fields = ["id", "title", "status"];
+
+        response.body = JSON.parse(response.body);
+
+        //Flattened array of project JsonObjects containing info about name, id and description, add any other info later as necessary
+        let extractProjectInfo: any[] = new Array();
+
+        //Iterate through all the projects and extract meta information in extractColumns Array having JSON Objects with keys of id, name and description,
+        _.forEach(body, (currentProject: { status: string; id: string; meta: { name: any; description: any; }; }) => {
```

Comment:
if we are not using lodash extensively and we can use the native java functions to do the same, please use the native functions

Reply:
Have removed lodash now. Using JS native function to extract values and iterate over the projects json structure